apps/site/next-data/generators/vulnerabilities.mjs

```diff
@@ -0,0 +1,40 @@
+/**
+ * Groups vulnerabilities by major version number extracted from the `vulnerable` string.
+ *
+ * @param {Array<import('#site/types/vulnerabilities').Vulnerability>} vulnerabilities Array of Vulnerability objects
+ */
+function groupVulnerabilitiesByMajor(vulnerabilities) {
+  const grouped = {};
+
+  for (const vulnerability of vulnerabilities) {
+    const majorVersions =
+      vulnerability.vulnerable
+        .match(/\b\d+\b/g)
+        ?.map(Number)
```

Comment:
This is an odd way of extracting this, I think you can extract the matcher to the assignment of majorVersions to make it cleaner